src/main/java/com/minecolonies/client/gui/WindowHutBuilder.java

```diff
@@ -1,38 +1,140 @@
      */
     private static final String HUT_BUILDER_RESOURCE_SUFFIX = ":gui/windowHutBuilder.xml";
+    private static final String LIST_RESOURCES              = "resources";
+    private static final String PAGE_RESOURCES              = "resourceActions";
+    private static final String BUTTON_PREVPAGE             = "prevPage";
+    private static final String BUTTON_NEXTPAGE             = "nextPage";
+
+    private static final String VIEW_PAGES                = "pages";
+
+    private Button             buttonPrevPage;
+    private Button             buttonNextPage;
+
+    private Map<String, Integer> resources;
+    private BuildingBuilder.View     builder;
 
     /**
-     * Constructor for window builder hut
+     * Constructor for window builder hut.
      *
-     * @param building {@link com.minecolonies.colony.buildings.BuildingBuilder.View}
+     * @param building {@link com.minecolonies.colony.buildings.BuildingBuilder.View}.
      */
-    public WindowHutBuilder(BuildingBuilder.View building)
+    public WindowHutBuilder(final BuildingBuilder.View building)
     {
         super(building, Constants.MOD_ID + HUT_BUILDER_RESOURCE_SUFFIX);
+        this.builder = building;
+        pullResourcesFromHut();
     }
 
     /**
-     * Returns the name of a building
+     * Retrieve resources from the building to display in GUI
+     */
+    private void pullResourcesFromHut()
+    {
+        if (builder.getColony().getBuilding(builder.getID()) != null)
+        {
+            resources = builder.getNeededResources();
+        }
+    }
+
+    /**
+     * Returns the name of a building.
      *
-     * @return Name of a building
+     * @return Name of a building.
      */
     @NotNull
     @Override
     public String getBuildingName()
     {
         return "com.minecolonies.gui.workerHuts.buildersHut";
     }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+        findPaneOfTypeByID(BUTTON_PREVPAGE, Button.class).setEnabled(false);
+
+        buttonNextPage = findPaneOfTypeByID(BUTTON_NEXTPAGE, Button.class);
+        buttonPrevPage = findPaneOfTypeByID(BUTTON_PREVPAGE, Button.class);
+        pullResourcesFromHut();
+
+        Object[] entries = resources.entrySet().toArray();
+
+        final ScrollingList resourceList = findPaneOfTypeByID(LIST_RESOURCES, ScrollingList.class);
+        resourceList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return resources.size();
+            }
+
+            @Override
+            public void updateElement(int index, @NotNull Pane rowPane)
+            {
+                Object obj = entries[index];
+
+                if (obj instanceof Map.Entry && ((Map.Entry) obj).getKey() instanceof String && ((Map.Entry) obj).getValue() instanceof Integer)
+                {
+                    @NotNull String key = (String) ((Map.Entry) obj).getKey();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'key' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)